packages/react-storage/src/components/StorageImage/StorageImage.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { isFunction } from '@aws-amplify/ui';
+import { Image, ComponentClassNames } from '@aws-amplify/ui-react';
+import { useStorageURL } from '@aws-amplify/ui-react/internal';
+
+import type { StorageImageProps } from './types';
+
+export const StorageImage = ({
+  accessLevel,
+  className,
+  fallbackSrc,
+  identityId,
+  imgKey,
+  onStorageGetError,
+  ...rest
+}: StorageImageProps): JSX.Element => {
+  const { url, error } = useStorageURL(
+    imgKey,
+    { level: accessLevel, identityId },
+    fallbackSrc
+  );
+
+  if (error && isFunction(onStorageGetError)) {
+    onStorageGetError(error);
+  }
```

Comment:
This should be in an effect right?

Reply:
Actually now that i think abt it, shouldn't the callback just be passed to `useStorageUrl`?

Reply:
Yeah, for sure. Lets remove the error from return and do it internally